### ISSUE TITLE
cannot apply migration

### ISSUE THREAD

**Issue by sdahlbac** [hasura/graphql-engine#9501]
### Version Information

Server Version: 2.20.0
CLI Version (for CLI related issue):2.20.0

### Environment

OSS

database is mssql

### How to reproduce the issue?

1. `hasura migrate apply`
2. failure

### Screenshots or Screencast

```
hasura migrate status --admin-secret ...--endpoint ...
? Select a database to use hasura

Database: hasura
VERSION        NAME      SOURCE STATUS  DATABASE STATUS
1678804432525  squashed  Present        Not Present
```

```
hasura migrate apply --admin-secret ... --endpoint ...
? Select a database to use hasura
ERRO skipping applying migrations on database 'hasura', encountered:
{
  "code": "unexpected",
  "error": "mssql internal error",
  "internal": {
    "error": "No active transaction exist; cannot commit"
  },
  "path": "$"
}
FATA[0004] applying migrations failed on database(s): hasura
```
### Keywords
mssql

**Comment by meetzaveri** [comment]
Hello @sdahlbac 

At present, we do not have MSSQL support for Hasura CLI. So as a result, you cannot apply migrations via `hasura migrate apply` command against Hasura which is connected to MSSQL data source.

There is an open GH issue which you can track - https://github.com/hasura/graphql-engine/issues/8181

**Comment by sdahlbac** [comment]
@meetzaveri that is interesting, since before squashing I added one new migration file, and could apply that one just fine !?

**Comment by sdahlbac** [comment]
Turns out it was actually a busted migration, specifically `CREATE INDEX "itAssetsApplicationsComponents_applicationId_index" ON "dbo"."IT_Assets_Applications_Components" (applicationId);` with a typoed table name. After fixing that I can apply migrations just fine against mssql.

**Comment by meetzaveri** [comment]
Right, I you can apply migrations but when it comes to creating dump, it only supports Postgres for now, i.e. CLI.  Updated my original comment for more info.
